src/js/_enqueues/admin/user-profile.js

```diff
@@ -332,6 +340,75 @@
 		}
 	}
 
+	/**
+	 * Binds Caps Lock detection to a given password input field.
+	 *
+	 * @param {jQuery} $input The password input field.
+	 */
+	function bindCapsLockWarning( $input ) {
+		var $capsWarning = $( '#caps-warning' ),
+			capsLockOn   = false;
+
```

Comment:
Per below, we can skip adding the `keydown` event when on a Mac to in the first place.

```suggestion

		// Skip warning on macOS Safari + Firefox (they show native indicators).
		if ( isMac && ( isSafari || isFirefox ) ) {
			return;
		}

```